src/artifactContentRenderers/artifactContentRendering.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as vscode from "vscode";
+import * as sarif from "sarif";
+import { BinaryArtifactContentRenderer } from "./binaryArtifactContentRenderer";
+import { TextArtifactContentRenderer } from "./textArtifactContentRenderer";
+import { RenderingArtifactContentRenderer } from "./renderingArtifactContentRenderer";
+
+export interface ArtifactContentRenderer {
+    /**
+     * Computes a range from the content it will render.
+     * @param region The region for the artifact content.
+     */
+    rangeFromRegion?(region: sarif.Region): vscode.Range | undefined;
+
+    render(artifactUri: vscode.Uri): string;
+
+    specificUriExtension?: string;
+}
+
+type tryCreateLogFile = (log: sarif.Log, artifactContents: sarif.ArtifactContent, runIndex: number, artifactIndex: number) => ArtifactContentRenderer | undefined;
```

Comment:
`log`, `runIndex`, `artifactIndex` unused.

Reply:
This is defining a type for line 25, it isn't implementing a function

---
In reply to: [411728013](https://github.com/microsoft/sarif-vscode-extension/pull/252#discussion_r411728013) [](ancestors = 411728013)